### PR TITLE
Chore/Remove PovioKit dependency

### PR DESCRIPTION
**Pull request by borut-t**
Completely removed reliance on the PovioKit dependency.

### CODE REVIEW

File: Sources/LinkedIn/Core/HttpClient.swift
Comment (yllfejziu) [inline Sources/LinkedIn/Core/HttpClient.swift:1]: Why is the HttpClient part of the LinkedIn folder? Can we move it out?

**Review by yllfejziu — COMMENTED** [review]
👌

**Review by yllfejziu — APPROVED** [review]

File: Sources/LinkedIn/Core/HttpClient.swift
Comment (borut-t) [reply Sources/LinkedIn/Core/HttpClient.swift:1]: It's internal implementation only needed by LinkedIn target.

File: Sources/LinkedIn/Core/HttpClient.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+//
+//  Created by Borut Tomazin on 22/05/2024.
+//  Copyright © 2024 Povio Inc. All rights reserved.
+//
+
+import Foundation
+
+struct HttpClient {
+  func request(method: String, url: URL, headers: [Header]) async throws -> Data {
+    var urlRequest = URLRequest(url: url)
```
Comment (tonikocjan) [inline Sources/LinkedIn/Core/HttpClient.swift:13]: I would move serialization and deserialization into this function.

File: Sources/LinkedIn/Core/HttpClient.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    urlRequest.httpMethod = method
+    headers.forEach {
```
Comment (tonikocjan) [inline Sources/LinkedIn/Core/HttpClient.swift:15]: You can set `allHTTPHeaderFields` directly (the type of `headers` needs to be adjusted).

Comment (borut-t) [reply Sources/LinkedIn/Core/HttpClient.swift:15]: What is the benefit here?

Comment (tonikocjan) [reply Sources/LinkedIn/Core/HttpClient.swift:15]: Now, an array of tuples is created, traversed, and converted into a dictionary.
Instead, you can create a dictionary and directly assign it to the request. 🤷‍♂️

File: Sources/LinkedIn/Core/HttpClient.swift (hunk 1 above)
Comment (borut-t) [reply Sources/LinkedIn/Core/HttpClient.swift:13]: I don't see any benefit by doing that.

File: Sources/LinkedIn/Core/HttpClient.swift (hunk 2 above)
Comment (borut-t) [reply Sources/LinkedIn/Core/HttpClient.swift]: https://github.com/poviolabs/PovioKitAuth/pull/30/commits/dd8a2bc491936eea0e76941dd04646702625f46b

File: Sources/LinkedIn/Core/HttpClient.swift (hunk 1 above)
Comment (borut-t) [reply Sources/LinkedIn/Core/HttpClient.swift:19]: https://github.com/poviolabs/PovioKitAuth/pull/30/commits/a3d953122be4fa969af23a13bc2b8621922112d8

**Review by tonikocjan — APPROVED** [review]

**Pull request merged**